static/src/javascripts/bootstraps/common.js

```diff
@@ -442,6 +444,14 @@ define([
 
             initShareCounts: function () {
                 shareCount.init();
+            },
+
+            initWeatherComponent: function () {
+                mediator.on('page:common:ready', function () {
+                    if (config.switches && config.switches.weather && detect.hasGeoLocation()) {
```

Comment:
think this conditional would be better living in the weather module itself, so it's self contained